docker/cloudbuild.yaml

```diff
@@ -0,0 +1,22 @@
+steps:
+# Build the main getsentry/sentry image
+- name: 'gcr.io/cloud-builders/docker'
+  args: [ 'build', '--build-arg', 'SOURCE_COMMIT=$COMMIT_SHA', '-t', 'gcr.io/$PROJECT_ID/sentry:$COMMIT_SHA', '-t', 'gcr.io/$PROJECT_ID/sentry:$SHORT_SHA', '-t', 'gcr.io/$PROJECT_ID/sentry:latest', '-f', './docker/Dockerfile', '.' ]
+# Derive the -onbuild variant
+- name: 'gcr.io/cloud-builders/docker'
+  args: [ 'build', '-t', 'gcr.io/$PROJECT_ID/sentry:$COMMIT_SHA-onbuild', '-t', 'gcr.io/$PROJECT_ID/sentry:$SHORT_SHA-onbuild', '-t', 'gcr.io/$PROJECT_ID/sentry:latest-onbuild', '-f', './docker/onbuild/Dockerfile', '.' ]
+# Smoke tests
+- name: 'gcr.io/cloud-builders/docker'
+  args: ['config', 'get', 'system.secret-key']
+  env:
+    - SENTRY_SKIP_BACKEND_VALIDATION=1
+    - SENTRY_SECRET_KEY=abc
+    - SENTRY_REDIS_HOST=localhost
+timeout: 1200s
+images:
+- 'gcr.io/$PROJECT_ID/sentry:$COMMIT_SHA'
+- 'gcr.io/$PROJECT_ID/sentry:$COMMIT_SHA-onbuild'
+- 'gcr.io/$PROJECT_ID/sentry:$SHORT_SHA'
+- 'gcr.io/$PROJECT_ID/sentry:$SHORT_SHA-onbuild'
+- 'gcr.io/$PROJECT_ID/sentry:latest'
```

Comment:
I forget how we do this elsewhere, but we tend to be careful with `latest` tags, since this happens on branch builds and all that. And we don't wanna write `latest` on anything other than master. I'd almost prefer we don't use "latest", but we use branch name, then this is called "master" or whatever.

Reply:
Yup, latest here is dangerous, especially with multiple branches using the same trigger. I may have to remove this and find a way to do that only when pushing to Docker Hub.